src/app/services/ErrorLoggerService.ts

```diff
@@ -15,12 +15,16 @@ export class ErrorLoggerService {
     public static setSingletonInstance(service: ErrorLoggerService) {
         ErrorLoggerService.singleton = MGPOptional.of(service);
     }
+    public static logErrorAndFail(component: string, message: string, data?: JSONValue): MGPValidation {
+        ErrorLoggerService.logError(component, message, data);
+        throw new Error(`${component}: ${message}`);
```

Comment:
the throw is discriminated, he deserve data as well :D !

Reply:
pas très utile le "Erreur: FooComponent: erreur bli (extra data: [Object])" non ? :D

Reply:
par nature, un JSONValue, ça a le potentiel d'ête cyclique ? Si ça ne l'est pas on sais juste faire JSON.stringify(data) !
Et au pire, console.log :p

Reply:
abawi, stringify fait donc !